pandas/tests/sparse/test_series.py

```diff
@@ -88,6 +88,18 @@ def setup_method(self, method):
         self.ziseries2 = SparseSeries(arr, index=index, kind='integer',
                                       fill_value=0)
 
+    def test_constructor_data_input(self):
+        # see gh-16905
+        test_dict = {1: 1}
+        test_index = [0, 1, 2]
+        test_series = pd.Series(test_dict, index=test_index)
+
+        arr = SparseSeries(test_dict)
+        assert arr.count() == len(test_dict)
+
+        arr = SparseSeries(test_series, index=test_index)
+        assert arr.count() == test_series.count()
```

Comment:
I think what you should do is construct your expected result via instantiation as a `Series` (i.e. pass the `dict` into the `Series` constructor).  Then pass the `Series` object into the `SparseSeries` constructor.  That becomes your expected output.